pkg/provider/consulcatalog/consul_catalog.go

```diff
@@ -294,6 +356,129 @@ func contains(values []string, val string) bool {
 	return false
 }
 
+func rootsWatchHandler(ctx context.Context, dest chan<- []string) func(watch.BlockingParamVal, interface{}) {
+	return func(_ watch.BlockingParamVal, raw interface{}) {
+		if raw == nil {
```

Comment:
Is this more of a legit expected case than the one L366? what I mean, is why do we log an error for the case L366, but not for this one? Shouldn't we do both or neither?

Reply:
Added a logmessage.